Update deployment guides to use volumes (#1559)
n8n-io/n8n-docs: docs/hosting/installation/server-setups/docker-compose.md

Create a `docker-compose.yml` file. Paste the following in the file:

```yaml
version: "3"
version: "3.7"

services:
  traefik:

>     image: "traefik"

Reviewer: I would prefer if we kept the Traefik example. No objections to providing a Caddy example in addition.

Author: Will change back to Traefik.

      - /home/<YOUR USERNAME>/n8n-local-files:/files
```

You will now be able to write files to the `/files` directory in n8n and they will appear on your server in either `/root/local-files` or `/home/<YOUR USERNAME>/n8n-local-files`, respectively.

### 6. Create `.env` file

Create a `.env` file and change it accordingly.

```bash

> # Replace <directory-path> with the path you will keep your Caddyfile in

Reviewer 2: ```suggestion
# Replace <directory-path> with the path to the directory containing your Caddyfile
```

SSL_EMAIL=user@example.com
```

### 7. Create a Caddyfile

> Create a `Caddyfile` in the `DATA_FOLDER` path you set in the `.env` file, Make sure you replace `<subdomain>` and `<domain_name>` with the same values you set in the `.env` file.

Reviewer 2: ```suggestion
Create a `Caddyfile` in the `DATA_FOLDER` path you set in the `.env` file. Make sure you replace `<subdomain>` and `<domain_name>` with the same values you set in the `.env` file.
```

```
<subdomain>.<domain_name> {
    reverse_proxy n8n:5678 {
      flush_interval -1
    }
}
```

### 7. Create the Docker volumes

> Create the Docker volume which is defined as `n8n_data`, In this volume, the database file from SQLite as well as the encryption key will be saved.

Reviewer 2: ```suggestion
Create the Docker volume that is defined as `n8n_data`. n8n will save the database file from SQLite and the encryption key in this volume.
```

```sh
docker volume create n8n_data
```

> We will also create a volume for the Caddy data, This is defined as `caddy_data`.

Reviewer 2: ```suggestion
Create a volume for the Caddy data, This is defined as `caddy_data`.
```
